database-management/migrations/20180325204028_remove-unused-ghost-tables.js

```diff
@@ -11,19 +11,24 @@ exports.up = async knex => {
     'permissions_apps',
     'permissions_roles',
     'permissions_users',
-    'roles',
     'refreshtokens',
+    'roles',
     'roles_users',
     'settings',
     'subscribers',
     'users',
     'clients',
   ];
-  for (tableName of tablesToDrop) {
+  // eslint-disable-next-line no-restricted-syntax
+  for (const tableName of tablesToDrop) {
+    // eslint-disable-next-line no-await-in-loop
```

Comment:
Makes sense, up to you but you may want to add a comment on the disables explaining why it is needed just to make it the standard that eslint-disables have to be justified. Also might be worth changing this to `tablesToDrop.forEach()` just to make this a little bit more javascript-y and minimize eslint disables. 

But not super opinionated on any of this as it's hardly critical so 👍

Reply:
Yeah it's all deliberate haha, `.forEach` doesn't work because it's a different function being run for each callback so it'll still run asynchronously and cause race conditions. Comments I can do though :). I think I added comments to at least one of the places I did something like that, but yeah I can definitely add it the other places too

Reply:
According to [this thread](https://github.com/tgriesser/knex/issues/148), Knex passes in Bluebird (a promise handling library) by default. Might be worth at looking at using [this](https://stackoverflow.com/questions/29546898/run-bluebird-promises-sequentially-without-return-values) Bluebird helper since the dependency is installed anyways.

Reply:
Though there is an argument that that is even more subtle than the weird, but native JS.

Reply:
Your call!

Reply:
Actually, on even further thought - why even use this loop and array at all? Each table to be dropped has to be manually written out in the array above anyways so is it not more idiomatic and eslint error free to get rid of this array and loop and just do 
```js
knex
.droptable()
.droptable()
.droptable()
...
```

Reply:
Either way, just semantics. This is totally fine as is.

Reply:
Quite a few comments since I last looked haha, I'll try to address most of them.

Yeah I know it passes in a Bluebird promise, as that's the default argument when you call `knex migrate:make` to make a new file, but yeah I definitely want to stick to native promises all around the codebase and not mix in bluebird, and also it creates a dependency on Knex as they could change how they supply that promise etc. so I don't really like that.

With regards to your dropTable chain is definitely an option, though I for some reason feel it would be a bit more ugly. As I was eating dinner what I was thinking of doing was actually just parallelizing it and putting in the minimal amount of work to figure out which order it is exactly they have to go in and then do the drops in 2-3 batches that are all parallelized, so I think I'll just do that actually